test/unit/src/common/containers/t_with-repository.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
+import withRepository from '../../../../../src/common/containers/with-repository';
+
+const middlewares = [ thunk ];
+const mockStore = configureMockStore(middlewares);
+
+// Dummy test component to wrap
+const Dummy = () => <span>Dummy</span>;
+
+import { SET_GITHUB_REPOSITORY } from '../../../../../src/common/actions/create-snap';
+
+describe('WithRepository', () => {
+  const PARAMS = {
+    owner: 'anowner',
+    name: 'aname'
+  };
+  const REPO = {
+    fullName: 'anowner/aname'
+  };
+
+  let store;
+
+  let wrapper;  // WithRepository rendered wrapper
+  let instance; // WithRepository wrapper instance
+  let wrapped;  // wrapped Dummy component
+
+  const setupDummy = (repository = REPO, params = PARAMS) => {
```

Comment:
Possibly it could, but what it actually prepares is `wrapper`, `instance` and `wrapped` objects that are used in tests.

If it only returns `wrapper`, each time we call setupDummy we would need to recreate those manually.

If that would help this helper function can be moved to the end of `describe`, so it will be kept in the scope of the variables it uses, but will not take space at the top of tests file.